src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,327 @@
+      List<? extends Collection<Term>> pPartitionedFormulas, int[] pStartOfSubTree)
+      throws SolverException, InterruptedException {
+    Preconditions.checkState(!closed);
+    assert InterpolatingProverEnvironment.checkTreeStructure(
+        pPartitionedFormulas.size(), pStartOfSubTree);
+
+    // Generate every Interpolation Pair
+    ImmutableList<ImmutableList<ImmutableList<Collection<Term>>>> interpolationPairs =
+        getTreeInterpolationPairs(pPartitionedFormulas, pStartOfSubTree);
+
+    final List<Term> itps = Stream.of(new Term[] {}).collect(Collectors.toList());
+    try { // Interpolate every Interpolation Pair
+      for (int i = 0; i < interpolationPairs.size(); i++) {
+        itps.add(
+            getCVC5Interpolation(
+                ImmutableList.copyOf(interpolationPairs.get(i).get(0)),
+                ImmutableList.copyOf(interpolationPairs.get(i).get(1))));
+      }
+    } catch (UnsupportedOperationException e) {
+      if (e.getMessage() != null) {
+        throw new SolverException(e.getMessage(), e);
+      } else {
+        throw e;
+      }
+    }
+
+    final List<BooleanFormula> result =
+        Stream.of(new BooleanFormula[] {}).collect(Collectors.toList());
+    for (Term itp : itps) {
+      result.add(creator.encapsulateBoolean(itp));
+    }
+    assert result.size() == pStartOfSubTree.length - 1;
+    return result;
+  }
+
+  /**
+   * Interpolates a Tuple of Interpolants according to Craig-Interpolation using CVC5-Interpolation.
+   *
+   * <p>CVC5's getInterpolant: There is a Model, such that the Interpolant I, with A -> I = True
+   * (1CVC5) and I -> B = True (2CVC5).
+   *
+   * <p>Craig Interpolation: There is a Model, such that the Interpolant psi, with phi- -> psi =
+   * True (1Craig), not(psi /\ phi+) = True (2Craig).
+   *
+   * <p>With A/phi- current set of assertions and B/phi+ Formulas to interpolate.
+   *
+   * <p>CVC5 -> Craig Interpolation:
+   *
+   * <p>(1CVC5) <=> (1Craig) due to subst of A with phi- and reflexivity.
+   *
+   * <p>(2CVC5) <=> I -> B = True <=> (not I) \/ B = True <=> not (I /\ (not B)) = True <=> (2Craig)
+   * due to subst of (not B) with phi+ and reflexivity.
+   *
+   * <p>Hence, CVC5's Interpolation produces an equivalent Interpolation to Craig Interpolation, if
+   * B is negated during CVC5 Interpolation.
+   *
+   * @param pFormAAsList current Set of Assertions A
+   * @param pFormBAsList Formulas to Interpolate B
+   * @return Interpolation of the Interpolation Pair following the definition of
+   *     Craig-Interpolation.
+   */
+  private Term getCVC5Interpolation(
+      ImmutableList<Collection<Term>> pFormAAsList, ImmutableList<Collection<Term>> pFormBAsList) {
+
+    // Uses a separate Solver instance to leave the original solver-context unmodified
+    Solver interpolationSolver = new Solver();
+
+    setSolverOptions(seed, solverOptions, interpolationSolver);
+
+    interpolationSolver.resetAssertions();
```

Comment:
Why do we need to reset a fresh solver instance?

Reply:
We do not. Forgot to remove it.